server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -124,7 +124,8 @@
                             .payload(payload.getPayload())
                             .build();
 
-        return Mono.fromRunnable(() -> putBlobFunction.putBlob(bucketName, blob))
+        return createBucketIfNotExist(bucketName)
```

Comment:
No! This is not what we agreed upon!

We agreed upon:
 - Creating the blob in the bucket
 - Catching exceptions about blob creation exception because bucket does not exist
 - Then if the bucket does not exist, create it....
 - ... and save the blob again (potentially catching exceptions again)

A test that your code do not pass is that one:

```
ConcurrentTestRunner.builder()
    .threadCount(5).operationCount(20)
    .operation((a, b) -> {
        if ((a + b) % 5 == 0) {
            blobStore.deleteBucket(BUCKET);
         } else {
            blobStore.save(BUCKET, RANDOM_BLOB); // Will fail with Http exception as the bucket will be concurrently deleted between the check and the blob-saving operation
        }
    })
   .runSuccessfullyWithin(TEN_MINUTES);
```

Also note that you add a read operation before **every write**, while in a normal situation we can be quite optimistic that the bucket actually exist.

Please implement what we agreed upon